test/unit/select_dropped_indexes_test.lua

```diff
@@ -0,0 +1,136 @@
+local select_plan = require('crud.select.plan')
+
+local select_conditions = require('crud.select.conditions')
+local cond_funcs = select_conditions.funcs
+
+local t = require('luatest')
+local g = t.group('select_dropped_indexes')
+
+local helpers = require('test.helper')
+
+g.before_all = function()
+    helpers.box_cfg()
+
+    local customers = box.schema.space.create('customers', {
+        format = {
+            {'id', 'unsigned'},
+            {'bucket_id', 'unsigned'},
+            {'name', 'string'},
+            {'age', 'unsigned'},
+            {'number_of_pets', 'unsigned'},
+            {'cars', 'array'},
+        },
+        if_not_exists = true,
+    })
+
+    customers:create_index('id', {
+        type = 'TREE',
+        parts = {'id'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('bucket_id', {
+        parts = {'bucket_id'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers:create_index('age_tree', {
+        type = 'TREE',
+        parts = {'age'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers:create_index('name_hash', {
+        type = 'HASH',
+        parts = {'name'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('age_hash', {
+        type = 'HASH',
+        parts = {'age'},
+        if_not_exists = true,
+    })
+
+    customers:create_index('name_tree', {
+        type = 'TREE',
+        parts = {'name'},
+        unique = false,
+        if_not_exists = true,
+    })
+
+    customers.index.name_hash:drop()
+    customers.index.age_hash:drop()
```

Comment:
You've copied this code from test that checks smth related to index type, thats why indexes are named `*_<hash,tree>`. But these tests aren't about it. I suggest naming indexes just `index1, index2, ...` because we want to test that removing index doesn't affect iterating over indexes, so these names are good IMHO.

Reply:
You can also name some of them `index_dropped` or smth like this.